### ISSUE TITLE
(avro) incorrect deserialization of `long` with new `AvroFactory`

### ISSUE THREAD

**Issue by LvR** [FasterXML/jackson-dataformats-binary#113]
Hi,

I've found that AvroFactory incorreclty deserialize some long values. For exemple the value -5580797459299185431L .

```java
    Long src = new Long(-5580797459299185431L);

    AvroMapper objectMapper = new AvroMapper(new AvroFactory());
    AvroSchema avroSchemaSrc = objectMapper.schemaFor(src.getClass());
    ObjectWriter writer = objectMapper.writer(avroSchemaSrc);
    byte[] data = writer.writeValueAsBytes(src);
    // data bytes are ok, same as native avro writer

    ObjectReader reader = objectMapper.readerFor(src.getClass()).with(avroSchemaSrc);
    Long dest= reader.readValue(data);

    assertThat(dest).isEqualTo(src);
```

Serialised data are good however, tested by comparison with native avro lib.
I've also tested the other implementation given in package com.fasterxml.jackson.dataformat.avro.apacheimpl, ApacheAvroFactory, and everything works well.

Tested with jackson-dataformat-avro 1.9.1

Full Junit test attached with native avro impl, ApacheAvroFactory and the buggy AvroFactory.

Cheers
LvR
[jacksondataformatsavrotest.zip](https://github.com/FasterXML/jackson-dataformats-binary/files/1379200/jacksondataformatsavrotest.zip)

**Comment by cowtowncoder** [comment]
Ok thank you for reporting this (and esp appreciate throroughness of checking) -- must be a problem with native decoder's byte handling, hope to fix for 2.9.2 (I assume you mean 2.9.1)

**Comment by LvR** [comment]
Yes, this is 2.9.1
